File: back/src/main/java/com/tdevilleduc/urthehero/back/service/IStoryService.java
package com.tdevilleduc.urthehero.back.service;

import com.tdevilleduc.urthehero.back.model.Story;
import com.tdevilleduc.urthehero.back.model.StoryDTO;

import java.util.List;
import java.util.Optional;

public interface IStoryService {

    boolean exists(Integer storyId);
    boolean notExists(Integer storyId);
    Optional<Story> findById(Integer storyId);
    List<Story> findAll();
    List<Story> findByPersonId(Integer personId);
    StoryDTO createOrUpdate(StoryDTO storyDto);
    void delete(Integer storyId);
}

File: back/src/main/java/com/tdevilleduc/urthehero/back/service/IStoryService.kt
package com.tdevilleduc.urthehero.back.service

import com.tdevilleduc.urthehero.back.model.Story
import com.tdevilleduc.urthehero.back.model.StoryDTO

interface IStoryService {
    fun exists(storyId: Int): Boolean
    fun notExists(storyId: Int): Boolean
    fun findById(storyId: Int): Story
    fun findAll(): MutableList<Story>
    fun findByPersonId(personId: Int): MutableList<Story>
    fun createOrUpdate(storyDto: StoryDTO): StoryDTO
    fun delete(storyId: Int)
}